frontend/src/metabase/visualizations/lib/apply_tooltips.js

```diff
@@ -1,286 +1,238 @@
-            //
-            // for series 0, i = 3
-            // rowIndex = (3 % 6) - 1 = 3 - 1 = 2
-            const seriesData = series[seriesIndex].data || {};
-            const rawCols = seriesData._rawCols;
-            const rows = seriesData && seriesData.rows;
-            const rowIndex = rows && (i % (rows.length + 1)) - 1;
-            const row = rowIndex != null && seriesData.rows[rowIndex];
-            const rawRow = row && row._origin && row._origin.row; // get the raw query result row
-            // make sure the row index we've determined with our formula above is correct. Check the
-            // x/y axis values ("key" & "value") and make sure they match up with the row before setting
-            // the data for the tooltip
-            if (rawRow && row[0] === d.data.key && row[1] === d.data.value) {
-              // rather than just append the additional values we'll just create a new `data` array.
-              // simply appending the additional values would result in tooltips whose order switches
-              // between different series.
-              // Loop over *all* of the columns and create the new array
-              data = rawCols.map((col, i) => {
-                // if this was one of the original x/y columns keep the original object because it
-                // may have the `isNormalized` tweak above.
-                if (col === data[0].col) {
-                  return data[0];
-                }
-                if (col === data[1].col) {
-                  return data[1];
-                }
-                // otherwise just create a new object for any other columns.
-                return {
-                  key: getColumnDisplayName(col),
-                  value: rawRow[i],
-                  col: col,
-                };
-              });
-            }
-          }
-
-          if (isBreakoutMultiseries) {
-            data.unshift({
-              key: getFriendlyName(card._breakoutColumn),
-              // Use series title if it's set
-              value: seriesTitle ? seriesTitle : card._breakoutValue,
-              // Don't include the column if series title is set (it's already formatted)
-              col: seriesTitle ? null : card._breakoutColumn,
-            });
-          }
-
-          data = _.uniq(data, d => d.col);
-
-          onHoverChange({
-            // for single series bar charts, fade the series and highlght the hovered element with CSS
-            index: isSingleSeriesBar ? -1 : seriesIndex,
-            // for area charts, use the mouse location rather than the DOM element
-            element: isArea ? null : this,
-            event: isArea ? d3.event : null,
-            data: data.length > 0 ? data : null,
-          });
+        .on("mousemove", function(d) {
+          const hovered = getClickHoverHelper(this, d);
+          onHoverChange(hovered);
         })
         .on("mouseleave", function() {
           if (!onHoverChange) {
```

Comment:
This isn't part of your changes, but can this test ever be satisfied, given that it's enclosed in an `if (onHoverChange)`-block?